spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java

```diff
@@ -113,7 +113,7 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 		}
 
 		// gh-838
-		if (!Boolean.TRUE.equals(map.get("active"))) {
+		if (map.containsKey("active") && !"true".equals(String.valueOf(map.get("active")))) {
```

Comment:
if Boolean.FALSE.equals(map.get("active")) thrown ....

Reply:
No, because "active" could also be null or a String instead of a false boolean...

Reply:
and you just want to throw the exception if it's false, no?

Reply:
I guess so, but taking String values into account. So would you prefer the following?
```suggestion
		if (map.containsKey("active") && "false".equals(String.valueOf(map.get("active")))) {
```

Reply:
I'm not sure. I'd keep it simple. We're facing an issue with it here.
To me feels like you wanna to throw an exception just if it was explicitly disabled. If you have something else there it should be fine.

Reply:
The only difference with this change, is that a null active value (or any arbitrary value other than false) would be considered as a valid token... which may be a bit risky? Checking if it's true is more restrictive and looks safer to me.